src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -4093,10 +4093,13 @@ int32_t NaClSysExecve(struct NaClAppThread *natp, char const *path, char *const
   /* set up environment, only do this if we initially were passed an environment*/
   NaClEnvCleanserCtor(&env_cleanser, 0);
   if (envp) {
-    /* count number of environment strings */
-    for (char **pp = sys_envp_ptr; *pp; ++pp) {
+  
+    /* Count amount of env from acquired NaCl pointer */
+    uint32_t *envcountptr = sys_envp_ptr;
+    while (envcountptr[new_envc] != NULL) {
```

Comment:
The previous version checked for the number of non-null char*s (64 bits) in the array. Your current version checks the number of non-null u_int32. Without being to familiar with the surrounding code, this raised some red flags. I just want to make sure that this does what you want it to do.

Reply:
Right. That's what was giving the error!